src/bin/client.rs

```diff
@@ -0,0 +1,217 @@
+                    match cmd_encode {
+                        Ok(_) => {
+                            let status = s.read_u8();
+                            match status {
+                                Ok(st) => {
+                                    if st == Cnv::OkPkg as u8 {
+                                        println!("Server still reachable.");
+                                    } else {
+                                        println!("Server is responding but INSANE!");
+                                        //maybe close connection? -> timeout
+                                    }
+                                },
+                                Err(_) => {
+                                    println!("Error reading ping-package");
+                                }
+                            }
+                        }
+                        Err(_) => {
+                            println!("Sending ping-message failed");
+                        }
+                    }
+                },
+                _ => {
+                    let cmd_encode = encode_into(&input, &mut s,
+                        SizeLimit::Bounded(1024));
+                    match cmd_encode {
+                        Ok(_) => {
+                            let status = s.read_u8();
+                            match status {
+                                Ok(st) => {
+                                    if st == Cnv::ResponsePkg as u8 {
+                                        // decode Response
+                                    } else {
+                                        println!("Unexpected return");
+                                        // try again
+                                    }
+                                },
+                                Err(_) => {
+                                    println!("Error reading response-package");
+                                }
+                            }
+                        }
+                        Err(_) => {
+                            println!("Sending command-package failed. Try again.");
+                        }
+                    }
+                }
+            }
+        }
+        Err(_) => {
+            println!("Sending command-header failed");
+        }
+    }
+    1
+}
+
+/// Receive greeting from server
+fn receive_greeting<R: Read>(buf: &mut R) -> u8 {
+    let status = buf.read_u8();
+    match status {
+        Ok(st) => {
+            if st != Cnv::GreetPkg as u8 {
+                println!("Communication mismatch. Try again later.");
+                return 1;
+            } else {
+                // read greeting
+                let greeting: Greeting;
+                let greet = decode_from::<R, Greeting>(buf,
+                    SizeLimit::Bounded(1024));
+                match greet {
+                    Ok(gr) => {
+                        greeting = Greeting::make_greeting(gr.protocol_version,
```

Comment:
Why is `greeting` declared 5 lines above? Why not say `let greeting = ...` here?